bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreCatalog.java

```diff
@@ -0,0 +1,400 @@
+      LOG.warn("No configuration was set, using the default environment Configuration");
+      this.conf = new Configuration();
+    }
+
+    LOG.info("Using BigQuery Metastore Iceberg Catalog: {}", inputName);
+
+    if (properties.containsKey(CatalogProperties.WAREHOUSE_LOCATION)) {
+      // Iceberg always removes trailing slash to avoid paths like "<folder>//data" in file systems
+      // like s3.
+      this.conf.set(
+          HIVE_METASTORE_WAREHOUSE_DIR,
+          LocationUtil.stripTrailingSlash(properties.get(CatalogProperties.WAREHOUSE_LOCATION)));
+    }
+
+    String fileIoImpl =
+        properties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.hadoop.HadoopFileIO");
+    this.fileIO = CatalogUtil.loadFileIO(fileIoImpl, properties, conf);
+
+    this.filterUnsupportedTables =
+        Boolean.parseBoolean(properties.getOrDefault(FILTER_UNSUPPORTED_TABLES, "false"));
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    return new BigQueryTableOperations(
+        client,
+        fileIO,
+        projectId,
+        // Sometimes extensions have the namespace contain the table name too, so we are forced to
+        // allow invalid namespace and just take the first part here like other catalog
+        // implementations do.
+        identifier.namespace().level(0),
+        identifier.name(),
+        conf);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier identifier) {
+    String locationUri = null;
+    DatasetReference datasetReference = toDatasetReference(identifier.namespace());
+    Dataset dataset = client.getDataset(datasetReference);
+    if (dataset != null && dataset.getExternalCatalogDatasetOptions() != null) {
+      locationUri = dataset.getExternalCatalogDatasetOptions().getDefaultStorageLocationUri();
+    }
+    return String.format(
+        "%s/%s",
+        Strings.isNullOrEmpty(locationUri)
+            ? datasetReference.getDatasetId()
+            : LocationUtil.stripTrailingSlash(locationUri),
+        identifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    validateNamespace(namespace);
+
+    return client.listTables(toDatasetReference(namespace), filterUnsupportedTables).stream()
+        .map(
+            table -> TableIdentifier.of(namespace.level(0), table.getTableReference().getTableId()))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    try {
+      TableOperations ops = newTableOps(identifier);
+      TableMetadata lastMetadata = ops.current();
+
+      client.deleteTable(toTableReference(identifier));
+
+      if (purge && lastMetadata != null) {
+        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+      }
+    } catch (NoSuchTableException e) { // Not catching a NoSuchIcebergTableException on purpose
```

Comment:
both comments seem rather unnecessary to have

Reply:
Done